clients/da-vinci-client/src/main/java/com/linkedin/davinci/repository/NativeMetadataRepositoryViewAdapter.java

```diff
@@ -0,0 +1,223 @@
+    }
+    if (!VeniceView.isViewStore(storeName)) {
+      return store;
+    }
+    // It's a view store, so we need to annotate the version(s) before returning
+    List<Version> storeVersions = store.getVersions();
+    String viewName = VeniceView.getViewNameFromViewStoreName(storeName);
+    Map<Integer, ReadOnlyStore.ReadOnlyVersion> viewVersionMap = new HashMap<>();
+    for (Version version: storeVersions) {
+      ViewConfig viewConfig = version.getViewConfigs().get(viewName);
+      if (viewConfig == null) {
+        // versions that do not contain the corresponding view name is omitted
+        continue;
+      }
+      viewVersionMap
+          .put(version.getNumber(), new ReadOnlyStoreView.ReadOnlyMaterializedViewVersion(version, viewConfig));
+    }
+    return new ReadOnlyStoreView(store, viewVersionMap);
+  }
+
+  @Override
+  public Store getStoreOrThrow(String storeName) throws VeniceNoStoreException {
+    Store store = getStore(storeName);
+    if (store != null) {
+      return store;
+    }
+    throw new VeniceNoStoreException(storeName);
+  }
+
+  @Override
+  public boolean hasStore(String storeName) {
+    return nativeMetadataRepository.hasStore(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public Store refreshOneStore(String storeName) {
+    return nativeMetadataRepository.refreshOneStore(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public List<Store> getAllStores() {
+    return nativeMetadataRepository.getAllStores();
+  }
+
+  @Override
+  public long getTotalStoreReadQuota() {
+    return nativeMetadataRepository.getTotalStoreReadQuota();
+  }
+
+  @Override
+  public void registerStoreDataChangedListener(StoreDataChangedListener listener) {
+    nativeMetadataRepository.registerStoreDataChangedListener(listener);
+  }
+
+  @Override
+  public void unregisterStoreDataChangedListener(StoreDataChangedListener listener) {
+    nativeMetadataRepository.unregisterStoreDataChangedListener(listener);
+  }
+
+  @Override
+  public int getBatchGetLimit(String storeName) {
+    return nativeMetadataRepository.getBatchGetLimit(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public boolean isReadComputationEnabled(String storeName) {
+    return nativeMetadataRepository.isReadComputationEnabled(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public void subscribe(String storeName) throws InterruptedException {
+    String internalStoreName = VeniceView.getStoreName(storeName);
+    nativeMetadataRepository.subscribe(internalStoreName);
+    getSubscribedStoreViews(internalStoreName).add(storeName);
+  }
+
+  @Override
+  public void unsubscribe(String storeName) {
+    String internalStoreName = VeniceView.getStoreName(storeName);
+    Set<String> subscribedStoreViews = getSubscribedStoreViews(internalStoreName);
+    subscribedStoreViews.remove(storeName);
+    if (subscribedStoreViews.isEmpty()) {
+      nativeMetadataRepository.unsubscribe(internalStoreName);
```

Comment:
There is a race condition here..
I think we can use `compute` method of `VeniceConcurrentHasmMap` here to make it thread safe.